source/neuropod/internal/neuropod_tensor.cc

```diff
@@ -151,6 +151,27 @@ void NeuropodTensor::assure_device_cpu() const
     }
 }
 
+void NeuropodTensor::assure_view_compatible_shape(const std::vector<int64_t> &requested_dims) const
+{
+    // Make sure all provided dimensions are positive
+    if (!std::all_of(requested_dims.begin(), requested_dims.end(), [](int64_t i) { return i > 0; }))
+    {
+        NEUROPOD_ERROR("All dimensions provided to `view` must be positive");
+    }
+
+    // Compute the total number of elements
+    auto new_num_elements = compute_num_elements(requested_dims);
+    auto num_elements     = get_num_elements();
+
+    if (new_num_elements != num_elements)
```

Comment:
If we so strictly check it, we do need requested_dims at all?

Do we expect that requested_dims can be ever different than Tensor's dims?

Reply:
We're checking that the number of elements match, not that the dims match.

In fact, we expect that `requested_dims` will almost always be different than the tensor's dims.

The purpose of `view` is to view a tensor as if it had a different set of dimensions. The major restriction here is that the total number of elements must be the same. 

For example, viewing a `(10, 10, 3)` tensor as a `(10, 30)` or `(1, 300)` or `(300,)` tensor is fine because the total number of elements is the same.

`view` is quite similar to `reshape` in numpy except it guarantees that a copy of the underlying data is not made (see https://numpy.org/doc/stable/reference/generated/numpy.reshape.html for more details). Also see the link to `view` in pytorch in the PR description

Reply:
I see, I expected that having required_dims should allow even if we have "smaller" window but I guess requirement for same "size" is fine.